membership-attribute-service/app/actions/CommonActions.scala

```diff
@@ -1,15 +1,25 @@
 package actions
 import akka.stream.Materializer
+import com.gu.identity.RedirectAdviceResponse
 import components.{TouchpointBackends, TouchpointComponents}
 import controllers.NoCache
 import play.api.mvc._
+
 import scala.concurrent.{ExecutionContext, Future}
 
+sealed trait HowToHandleRecencyOfSignedIn
+case object Return401IfNotSignedInRecently extends HowToHandleRecencyOfSignedIn
+case object ContinueRegardlessOfSignInRecency extends HowToHandleRecencyOfSignedIn
+
 class CommonActions(touchpointBackends: TouchpointBackends, bodyParser: BodyParser[AnyContent])(implicit ex: ExecutionContext, mat:Materializer) {
   def noCache(result: Result): Result = NoCache(result)
 
   val NoCacheAction = resultModifier(noCache)
   val BackendFromCookieAction = NoCacheAction andThen new WithBackendFromCookieAction(touchpointBackends)
+  def AuthAndBackendViaIdapiAction(howToHandleRecencyOfSignedIn: HowToHandleRecencyOfSignedIn) =
+    NoCacheAction andThen new AuthAndBackendViaIdapiAction(touchpointBackends, howToHandleRecencyOfSignedIn)
+  def AuthAndBackendViaIdapiAction: ActionBuilder[AuthAndBackendRequest, AnyContent] =
```

Comment:
My personal preference would be to omit this utility `def` and instead use `AuthAndBackendViaIdapiAction(Return401IfNotSignedInRecently)`. 

Pros: clearer what's going on in-line; cons: more verbose. If you did want a utility def, I'd advocate for referencing how we are handling a user not being signed in recently in the action builder name.

Reply:
good point - done in https://github.com/guardian/members-data-api/pull/384/commits/1ff58275001f05802ed8dfe034a0e36e40cbbafe